subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -325,6 +325,18 @@ public Task findByPath(String path) {
         return new TaskLookupProvider<T>(type, name);
     }
 
+    @Override
+    public void configureLater(final String name, final Action<? super Task> action) {
```

Comment:
Should add some unit test coverage for this, as we'll be refactoring `DefaultTaskContainer` a fair bit. Should at least verify that calling this method does not realize the task and that the action is not executed until the task is realized.

This method will also need some error handling, maybe as a separate issue. We shouldn't defer this too long, though.